ttl/ttl.go

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) 2024 Hemi Labs, Inc.
+// Use of this source code is governed by the MIT License,
+// which can be found in the LICENSE file.
+
+package ttl
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"time"
+)
+
+var ErrNotFound = errors.New("not found")
+
+type value struct {
+	value any
+
+	expired func(any, any) // called when expired
+	remove  func(any, any) // called when removed
+
+	// Value context
+	ctx    context.Context
+	cancel context.CancelFunc
+}
+
+type TTL struct {
+	mtx sync.Mutex
+
+	m map[any]value
+}
+
+func New(capacity int) (*TTL, error) {
+	return &TTL{
+		m: make(map[any]value, capacity),
+	}, nil
+}
+
+func (tm *TTL) ttl(ctx context.Context, key any) {
+	select {
+	case <-ctx.Done():
```

Comment:
I don't think this `select` block is necessary as there is only one case, and could be simplified:
```suggestion
<-ctx.Done()
```